### PR TITLE
test: goldilocks test coverage

### PR DESCRIPTION
**Pull request by jorgeantonio21**

### CODE REVIEW

**Review by npwardberkeley — APPROVED** [review]

File: goldilocks/src/lib.rs [hunk 1]
```diff
@@ -321,3 +321,86 @@ unsafe fn add_no_canonicalize_trashing_input(x: u64, y: u64) -> u64 {
...
+    use super::*;
+
+    type F = Goldilocks;
+
+    #[test]
+    fn test_goldilocks() {
+        let f = F::new(100);
+        assert_eq!(f.as_canonical_u64(), 100);
+
+        let f = F::new(u64::MAX);
```
Comment (npwardberkeley) [inline goldilocks/src/lib.rs:336]: Maybe a quick comment about the reduction that makes these two equal?

**Review by npwardberkeley — APPROVED** [review]

File: goldilocks/src/lib.rs [hunk 2]
```diff
@@ -321,3 +321,86 @@ unsafe fn add_no_canonicalize_trashing_input(x: u64, y: u64) -> u64 {
...
+
+        let f = F::from_canonical_u64(F::ORDER_U64);
+        assert!(f.is_zero());
+
+        assert_eq!(
+            F::multiplicative_group_generator().as_canonical_u64(),
+            7_u64
+        );
+
+        let f_1 = F::new(1);
```
Comment (npwardberkeley) [inline goldilocks/src/lib.rs:353]: Oh also can we give these guys more meaningful names? Like `f_1` and `f_1_copy` for these two that are both `1`; `f_p_minus_1` for `f_4`, and so on?

Comment (jorgeantonio21) [reply goldilocks/src/lib.rs:353]: Refactored ! Thanks for the suggestions

File: goldilocks/src/lib.rs (hunk 1 above)
Comment (jorgeantonio21) [reply goldilocks/src/lib.rs:340]: Added a short comment on it

**Review by jorgeantonio21 — COMMENTED** [review]
Addressed the previous comments.

**Pull request merged**